llmfoundry/callbacks/hf_checkpointer.py

```diff
@@ -733,21 +803,44 @@ def tensor_hook(
                         monitor_process = None
 
                     # Spawn a new process to register the model.
-                    process = SpawnProcess(
-                        target=_register_model_with_run_id_multiprocess,
-                        kwargs={
-                            'mlflow_logger':
-                                mlflow_logger,
-                            'composer_logging_level':
-                                logging.getLogger('composer').level,
-                            'model_uri':
-                                local_save_path,
-                            'name':
-                                self.mlflow_registered_model_name,
-                            'await_creation_for':
-                                3600,
-                        },
-                    )
+                    # Slower method to register the model via log_model.
```

Comment:
As discussed offline, the call to `self.pre_register_edit` a few lines up assumes that there is an mlflow model at `local_save_path`. This is no longer true for non peft. So `self.pre_register_edit` needs to be called on `temp_save_dir` (which will contain the hf checkpoint, that log_model will copy over and register)